lib/targets/cuda/malloc.cpp

```diff
@@ -359,6 +368,61 @@ namespace quda
 #endif
     return ptr;
   }
+  /**
+   * Allocate shemm device memory. This function should only be called via
+   * device_comms_pinned_malloc_()
+   */
+#ifdef NVSHMEM_COMMS
+  void *shmem_malloc_(const char *func, const char *file, int line, size_t size)
+  {
+    MemAlloc a(func, file, line);
+
+    a.size = a.base_size = size;
+
+    auto dev = 0;
+    auto err = cudaGetDevice(&dev);
+    if (err != cudaSuccess) {
+      printfQuda("ERROR: cudaGetDevice failed in shmem_malloc_ (%s:%d in %s())\n", file, line, func);
+      errorQuda("Aborting");
+    }
+
+    if (!has_texture_alignment[dev]) {
```

Comment:
Should we be setting `texture_alignment` and `has_texture_alignment` for all devices as opposed to just device 0? If not, can we make these values scalars instead of arrays?

Reply:
guess with texture gone from QUDA we no longer need that at all ...